assets/js/view/part/towns.rb

```diff
@@ -3,17 +3,38 @@
 require 'snabberb/component'
 
 require 'view/part/town_dot'
+require 'view/part/town_rect'
 
 module View
   module Part
     class Towns < Snabberb::Component
+      ROUTE_COLORS = %i[red blue green purple].freeze
+
       needs :tile
       needs :region_use
+      needs :routes
 
       def render
-        @tile.towns.map do |_|
-          h(Part::TownDot, region_use: @region_use) if @tile.lawson?
+        @tile.towns.map do |town|
+          if @tile.lawson?
+            h(Part::TownDot, region_use: @region_use, color: color_for(town))
+          else
+            paths = @tile.paths.select { |p| p.town == town }
```

Comment:
maybe we should extract this pattern to be more efficient

Reply:
Something like this? ecb2dde

Reply:
(added 1e13ec1)

Reply:
hmm a bit fancier than i expected.

Reply:
i guess the caching is nice

Reply:
What approach were you thinking of for improving the efficiency? It would seem I didn't come up with the same approach you had in mind--I'm pretty sure apart from the caching, what I did was just as efficient :grinning:

Reply:
yea let's keep for now, maybe i'll come up with something later